scipy/stats/tests/test_continuous.py

```diff
@@ -1237,6 +1237,12 @@ def moment(self, order, kind='raw', *, a, b):
 
         X = LogUniform(a=np.exp(1), b=np.exp(3))
         Y = stats.exp(Uniform(a=1., b=3.))
+
+        # pre-2.0 support is not needed for much longer, so let's just test with 2.0+
+        if np.__version__ >= "2.0":
+            assert str(Y) == "exp(Uniform(a=1.0, b=3.0))"
+            assert repr(Y) == "exp(Uniform(a=np.float64(1.0), b=np.float64(3.0)))"
```

Comment:
To add the desired coverage I think the test needs to be for `X` not `Y`
```suggestion
            assert str(X) == "MyLogUniform(a=2.718281828459045, b=20.085536923187668)"
            assert repr(X) == "MyLogUniform(a=np.float64(2.718281828459045), b=np.float64(20.085536923187668))"
```

Reply:
Thanks @j-bowhay. I had that, but then tried to save space without thinking. I changed `a` and `b` to keep things a bit more compact.